server/services/openAIService.js

```diff
@@ -147,85 +151,122 @@ const evaluateAnswer = async (interviewText) => {
 
   const response = await callOpenAIAPI(messages, model, maxTokens);
   const processedResponse = processAPIResponse(response, model);
-  console.log("Processed response:", processedResponse);
 
   return processedResponse.result;
 };
 
-const evaluateTranscriptionForAllQuestions = async (userId, jobId) => {
+const updateOrCreateResult = async (
+  interviewId,
+  questionId,
+  scores,
+  feedback = ""
+) => {
+  let result = await ResultService.findResultByInterviewId(interviewId);
+
+  const questionData = {
+    question_id: questionId,
+    scores: scores,
+    feedback: feedback,
+  };
+
+  if (!result) {
+    await Result.create({
+      interview_id: interviewId,
+      question_scores: [questionData],
+    });
+  } else {
+    result.question_scores.push(questionData);
+    await result.save();
+  }
+};
+
+const evaluateTranscriptionForAllQuestions = async (
+  interviewId,
+  questionId,
+  transcription
+) => {
+  if (!questionId) {
+    console.log("Skipping empty question or answer");
+    return;
+  }
+
+  if (!transcription || transcription === "") {
+    let defaultScores = {
+      communication_skills: 0,
+      subject_expertise: 0,
+      relevancy: 0,
+    };
+    await updateOrCreateResult(interviewId, questionId, defaultScores);
+    return;
+  }
+
+  const questionDoc = await Question.findById(questionId);
+
+  const question = questionDoc.question;
+
+  const response = await evaluateAnswer({ question, answer: transcription });
+
+  const scores = response.scores;
+  const feedback = response.feedback;
+
+  await updateOrCreateResult(interviewId, questionId, scores, feedback);
```

Comment:
Why not evaluate for each question and answer, and then update all of them together. Try reducing the number of DB operations.

Reply:
I have used the existing code. If you want me to change I will change it.